xgcm/grid_ufunc.py

```diff
@@ -593,8 +612,36 @@ def apply_as_grid_ufunc(
     if grid is None:
         raise ValueError("Must provide a grid object to describe the Axes")
 
-    if any(not isinstance(arg, xr.DataArray) for arg in args):
-        raise TypeError("All data arguments must be of type DataArray")
+    # TODO: we need some more comprehensive tests that check proper padding with
+    # TODO: e.g. multiple vector components or mixed (tracer, vector components)
+    if not all(isinstance(a, xr.DataArray) or isinstance(a, dict) for a in args):
+        raise TypeError(
+            "All data arguments must be of type DataArray or dict (for vector components"
+        )
+
+    def _parse_and_check_other_component(
```

Comment:
Can we think of a better name for this function?

Reply:
how about `ensure_matching_other_components`?